### PR TITLE
Embed all binary textures

### PR DESCRIPTION
**Pull request by bagnell**
Embed all binary textures. Only check if they are supported by Jimp when using AO.

### CODE REVIEW

File: lib/loadGltfUris.js [hunk 1]
```diff
@@ -115,9 +115,10 @@ function isTransparent(image) {
 function readImageFromFile(object, name, uriPath) {
     return fsReadFile(uriPath)
         .then(function(data) {
+            var ext = path.extname(uriPath);
             object.extras._pipeline.source = data;
-            object.extras._pipeline.extension = path.extname(uriPath);
-            if (name === 'images') {
+            object.extras._pipeline.extension = ext;
+            if (name === 'images' && ext !== '.ktx' && ext !== '.gif') {
```
Comment (lilleyse) [inline lib/loadGltfUris.js]: Should this check stay in `generateJimpImage`? If the image is embedded it will not reach this code.

Comment (bagnell) [inline lib/loadGltfUris.js]: Generating a binary glTF reaches this code.

Comment (lilleyse) [inline lib/loadGltfUris.js]: I mean when the gltf is loaded it won't get there.

Comment (lilleyse) [inline lib/loadGltfUris.js]: Like this file with an embedded gif - Jimp will try to load it but fail: 
[BoxTextured-optimized.gltf.txt](https://github.com/AnalyticalGraphicsInc/gltf-pipeline/files/662235/BoxTextured-optimized.gltf.txt)

Comment (bagnell) [inline lib/loadGltfUris.js]: Ah, thanks. Updated.

**Comment by lilleyse** [comment]
Looks good - and now gif/ktx can actually go through the pipeline safely without failing (unless using AO).

**Pull request merged**